### PR TITLE
Make default ACL server-based

### PR DESCRIPTION
**Pull request by b123400**
connect https://github.com/SkygearIO/skygear-server/issues/309

Client side needs update as well
- JS https://github.com/SkygearIO/skygear-SDK-JS/pull/180
- iOS https://github.com/SkygearIO/skygear-SDK-iOS/pull/83

### CODE REVIEW

**Comment by oursky-travis** [comment]
@b123400, thanks for your PR! By analyzing the history of the files in this pull request, we identified @ben181231, @cheungpat and @limouren to be potential reviewers.

**Comment by cheungpat** [comment]
@b123400 Make sure that your PR title and comment is public consumable because the PR title and comment is usually included in the merge commit.

Specifically “a server side thing” is not good for public understanding, though I totally understand what you are talking about.

File: pkg/server/handler/schema.go [hunk 1]
```diff
@@ -497,3 +498,104 @@ func (h *SchemaAccessHandler) Handle(rpayload *router.Payload, response *router.
...
+func (payload *schemaDefaultAccessPayload) Decode(data map[string]interface{}) skyerr.Error {
+	if err := mapstructure.Decode(data, payload); err != nil {
+		return skyerr.NewError(skyerr.BadRequest, "fails to decode the request payload")
+	}
+
+	acl := skydb.RecordACL{}
+	for _, v := range payload.RawDefaultAccess {
+		ace := skydb.RecordACLEntry{}
+		if err := (*skyconv.MapACLEntry)(&ace).FromMap(v); err != nil {
+			return skyerr.NewInvalidArgument("invalid _default_access entry", []string{"_default_access"})
```
Comment (ben181231) [inline pkg/server/handler/schema.go]: The argument name should be `default_access`, right?

File: pkg/server/skydb/pq/migration/full.go [hunk 2]
```diff
@@ -122,6 +122,12 @@ CREATE TABLE _record_creation (
     FOREIGN KEY (role_id) REFERENCES _role(id)
 );
 CREATE INDEX _record_creation_unique_record_type ON _record_creation (record_type);
+CREATE TABLE _record_default_access (
+    record_type text NOT NULL,
+    default_access jsonb,
+    UNIQUE (record_type, default_access)
```
Comment (ben181231) [inline pkg/server/skydb/pq/migration/full.go]: The unique constraint should be on `record_type` only ?

**Comment by ben181231** [comment]
@b123400 @rickmak @cheungpat Do we need a default ACL for **all** record types ?

**Comment by b123400** [comment]
Updated to fix the issue

**Pull request merged**